pkg/provisioner/resourcehandlers.go

```diff
@@ -292,13 +280,33 @@ func updateObjectBucketClaimPhase(c versioned.Interface, obc *v1alpha1.ObjectBuc
 }
 
 func updateObjectBucketPhase(c versioned.Interface, ob *v1alpha1.ObjectBucket, phase v1alpha1.ObjectBucketStatusPhase, retryInterval, retryTimeout time.Duration) (result *v1alpha1.ObjectBucket, err error) {
-	logD.Info("updating status:", "ob", ob.Name, "old status", ob.Status.Phase,
-		"new status", phase)
+	logD.Info("updating objectBucket status:", "ob", ob.Name, "current phase", ob.Status.Phase,
+		"next phase", phase)
 	ob.Status.Phase = phase
 
 	err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
 		result, err = c.ObjectbucketV1alpha1().ObjectBuckets().UpdateStatus(ob)
-		return (err == nil), err
+		return err == nil, err
 	})
 	return
 }
+
+func validateObjectBucket(ob *v1alpha1.ObjectBucket) error {
+	var errs []string
+	if ob.Spec.Endpoint == nil {
+		errs = append(errs, "Spec.Endpoint cannot be nil")
+	}
+	if ob.Spec.Authentication == nil {
+		errs = append(errs, "Spec.Authentication cannot be nil")
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("%s", strings.Join(errs, "; "))
+	}
+	return nil
+}
+
+func setClaimOwnership(obc *v1alpha1.ObjectBucketClaim, labels map[string]string, finalizers []string) *v1alpha1.ObjectBucketClaim {
```

Comment:
maybe func name should be more generic? The name implies only ownership is set but labels are too.